src/cluster.c

```diff
@@ -5832,9 +5832,7 @@ clusterNode *getNodeByQuery(client *c, struct redisCommand *cmd, robj **argv, in
              * cluster is down. */
             if (error_code) *error_code = CLUSTER_REDIR_DOWN_STATE;
             return NULL;
-        } else if ((cmd->flags & CMD_WRITE) && !(cmd->proc == evalCommand)
-                && !(cmd->proc == evalShaCommand))
-        {
+        } else if (cmd->flags & CMD_WRITE) {
```

Comment:
The comment below here isn't quite right anymore.

Reply:
Comments adjusted.

Reply:
This code seems like a bug to me. If we have a multi-exec with write commands, and the cluster transitions to DOWN while we have "cluster_allow_reads_when_down" enabled, I think the exec would be allowed, even though it shouldn't? Anyways, I'll look int this more.